ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/ClusterTerminationTest.java

```diff
@@ -55,48 +57,103 @@ public class ClusterTerminationTest {
 
   private static final String SINK_TOPIC = "sink_topic";
   private static final String SINK_STREAM = "sink_stream";
+  private static final String ALL_TOPICS = ".*";
 
   private static final IntegrationTestHarness TEST_HARNESS = IntegrationTestHarness.build();
 
-  private static final TestKsqlRestApp REST_APP = TestKsqlRestApp
+  private static final TestKsqlRestApp REST_APP_0 = TestKsqlRestApp
       .builder(TEST_HARNESS::kafkaBootstrapServers)
       .withStaticServiceContext(TEST_HARNESS::getServiceContext)
       .withProperty(KsqlConfig.SCHEMA_REGISTRY_URL_PROPERTY, "http://foo:8080")
+      .withProperty(KsqlRestConfig.INTERNAL_LISTENER_CONFIG, "http://localhost:8088")
+      .withProperty(KsqlRestConfig.ADVERTISED_LISTENER_CONFIG, "http://localhost:8088")
+      .build();
+
+  private static final TestKsqlRestApp REST_APP_1 = TestKsqlRestApp
+      .builder(TEST_HARNESS::kafkaBootstrapServers)
+      .withStaticServiceContext(TEST_HARNESS::getServiceContext)
+      .withProperty(KsqlConfig.SCHEMA_REGISTRY_URL_PROPERTY, "http://foo:8080")
+      .withProperty(KsqlRestConfig.INTERNAL_LISTENER_CONFIG, "http://localhost:8089")
+      .withProperty(KsqlRestConfig.ADVERTISED_LISTENER_CONFIG, "http://localhost:8089")
       .build();
 
   @ClassRule
   public static final RuleChain CHAIN = RuleChain
-      .outerRule(Retry.of(3, ZooKeeperClientException.class, 3, TimeUnit.SECONDS))
-      .around(TEST_HARNESS)
-      .around(REST_APP);
+      .outerRule(Retry.of(3, ZooKeeperClientException.class, 3, TimeUnit.SECONDS));
+
+  @Rule
+  public final RuleChain CHAIN_TEST = RuleChain
+      .outerRule(TEST_HARNESS)
+      .around(REST_APP_0)
+      .around(REST_APP_1);
 
-  @BeforeClass
-  public static void setUpClass() {
+  @Before
+  public void setUp() {
```

Comment:
Can we expand this setup to include a query with intermediate internal topics? That way we can distinguish between the tests where we just clean up sink topics, vs the .* test.

Reply:
added